packages/db/src/repositories/perps.ts

```diff
@@ -1415,67 +1445,110 @@ export class PerpsRepository {
   }
 
   /**
-   * Get risk metrics leaderboard for a competition
+   * Batch create risk metrics snapshots for time series data
+   * Following the same pattern as batchCreatePortfolioSnapshots
+   * @param snapshots Array of risk metrics snapshot data
+   * @returns Array of created snapshots
+   */
+  async batchCreateRiskMetricsSnapshots(
+    snapshots: InsertRiskMetricsSnapshot[],
+    tx?: Transaction,
+  ): Promise<SelectRiskMetricsSnapshot[]> {
+    if (snapshots.length === 0) {
+      return [];
+    }
+
+    const executor = tx || this.#db;
+
+    try {
+      this.#logger.debug(
+        `[PerpsRepository] Batch creating ${snapshots.length} risk metrics snapshots`,
+      );
+
+      const now = new Date();
+      const results = await executor
+        .insert(riskMetricsSnapshots)
+        .values(
+          snapshots.map((snapshot) => ({
+            ...snapshot,
+            timestamp: snapshot.timestamp || now,
+          })),
+        )
+        .returning();
+
+      this.#logger.debug(
+        `[PerpsRepository] Successfully created ${results.length} risk metrics snapshots`,
+      );
+
+      return results;
+    } catch (error) {
+      this.#logger.error("Error in batchCreateRiskMetricsSnapshots:", error);
+      throw error;
+    }
+  }
+
+  /**
+   * Get risk metrics time series data for a competition
    * @param competitionId Competition ID
-   * @param limit Optional limit
-   * @param offset Optional offset
-   * @returns Object with metrics array and total count
+   * @param agentId Optional agent ID to filter by
+   * @param startDate Optional start date
+   * @param endDate Optional end date
+   * @param limit Optional limit for the number of snapshots to return
+   * @param offset Optional offset for pagination
+   * @returns Array of risk metrics snapshots
    */
-  async getCompetitionRiskMetricsLeaderboard(
+  async getRiskMetricsTimeSeries(
```

Comment:
imo since the number of params is now > 3, we should make the optional params a typed object

Reply:
Also updated in [9ae396c](https://github.com/recallnet/js-recall/pull/1498/commits/9ae396c0ae77472da381b0c96846bcc2b469ec24)